src/site/siteConfigPropTypes.js

```diff
@@ -21,105 +21,161 @@ const settingsConfigPropType = PropTypes.shape({
 });
 
 /**
- * Callback for showing the side menu
+ * Additional Sites that can be switched to.
  */
-const capabilitiesPropType = PropTypes.object;
+const sitesPropType = PropTypes.arrayOf(PropTypes.shape({
+  /**
+   * The path to the site
+   */
+  path: PropTypes.string,
+
+  /**
+   * The title of the site
+   */
+  title: PropTypes.string,
+
+  /**
+   * The url to navigate to the site
+   */
+  url: PropTypes.string,
+}));
 
 /**
- * Individual menu item for side nav
+ * Describes the extensions
  */
-const menuItemPropType = PropTypes.arrayOf(PropTypes.shape({
+const extensionItemsPropType = PropTypes.arrayOf(PropTypes.shape({
   /**
-   * Text for the menu item
+   * The react key for the extension.
+   */
+  key: PropTypes.string,
+  /**
+   * The text describing the extension.
    */
   text: PropTypes.string,
+  /**
+   * The icon to represent the extension.
+   */
+  icon: PropTypes.elementType,
+  /**
+   * The modal to launch from the extension.
+   */
+  modal: PropTypes.elementType,
+}));
 
+/**
+ * Describes page content
+ */
+const pageContentConfigPropType = PropTypes.shape({
   /**
-   * On-click the menu item will take you here
+   * The path to the page
    */
   path: PropTypes.string,
+  /**
+   * Text for the page
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * The type of the content being loaded
+   */
+  type: PropTypes.string.isRequired,
+});
 
+/**
+ * Describes navigation config
+ */
+const navigationConfigPropType = PropTypes.arrayOf(PropTypes.shape({
+  /**
+   * The text for the nav item
+   */
+  label: PropTypes.string.isRequired,
   /**
-   * Sub menu items
+   * The singular page config item
    */
-  childItems: PropTypes.arrayOf(PropTypes.object),
+  pageConfig: pageContentConfigPropType,
+  /**
+   * children
+   */
+  children: PropTypes.array,
 }));
 
 /**
- * Menu items for side navigation
+ * Path to imports map
+ */
+const contentImportsPropType = PropTypes.object;
+/**
+ * Path to page config map
  */
-const menuItemsPropType = PropTypes.PropTypes.objectOf(menuItemPropType);
+const pageConfigPropType = PropTypes.object;
+/**
+ * Short routes to content routes.
+ */
+const routesMapPropType = PropTypes.object;
```

Comment:
What is the routesMap shape?

Reply:
It's a map of a random key to an object. I didn't look very hard but didn't see a clean way to represent it with prop types.